### ISSUE TITLE
[Feature Request] CBA settings for custom radio sounds

### ISSUE THREAD

**Issue by thePier** [michail-nikolaev/task-force-arma-3-radio#1487]
When TFAR updates, custom sounds (added by the player for replacing the default ones) are overwritten.
The request would be to allow the player to specify a custom sound file in CBA settings to replace the default ones from TFAR.

Since the player already went down the rabbit hole to find and modify specific sounds, a brutal copy/paste of the audio file path would be nothing.

Doing this way, the original sound files added from TFAR would remain untouched and could be used as a "fail over" in case there is some issue (eg. file not found) with the custom ones.

**Comment by Grezvany13** [comment]
Unfortunately Arma 3 requires sounds to be in the CfgSounds class, so using a string doesn't work.

However it should be possible to have a CfgSounds object in a mod or mission which extends the default sounds, and with a custom variable (eg. `tf_click = 1` and `tf_push = 1`) so they can be filtered.

**Comment by severgun** [comment]
@Grezvany13 I think he is talking about sounds in TS (radio-sounds folder). They have nothing related to arma configs.
There should be just dedicated folder for custom sounds which will NOT be overwritten on plugin update. Even CBA settings not required IMHO. Just check existance of files with proper naming in this folder and override default.

**Comment by dedmen** [comment]
> Just check existance of files with proper naming in this folder and override default.
You mean like a `radio-sounds-alt` folder, which TFAR will automatically use if it exists?
@thePier does that sound good to you?
Removes the need to add a seperate setting.

**Comment by severgun** [comment]
> You mean like a `radio-sounds-alt` folder, which TFAR will automatically use if it exists?
Yes. I see it like this. maybe `custom-radio-sounds` / `radio-sounds-custom` / any other suitable name
But it is my opinion.
With string from CBA settings users can paste unknown text and you need to parse and handle this.

**Comment by thePier** [comment]
A second folder to place the custom user sounds I feel it's a good compromise to avoid too much fiddling with the CBA settings.

You could add a CBA settings to toggle between the default as the custom ones just to avoid weird combinatioms, but idk on this

**Comment by dedmen** [comment]
> You could add a CBA settings to toggle between the default as the custom ones

As you said before
> Since the player already went down the rabbit hole to find and modify specific sounds

I think the player can handle that.

Need feedback on the folder name.
I think `radio-sounds-custom` would best fit the current style?

**Comment by thePier** [comment]
That name sounds (ha-ha) perfect to me. 

Maybe you could add a readme file stating the expected file names and/or specific audio caratteristics?

**Comment by dedmen** [comment]
Yeah we need some wiki stuff.
Maybe I'll just make a wiki page for custom sound files on github. Can also link the old 0.9 soundfiles which were removed in 1.0
